### PR TITLE
 DOC: Spellcheck of gotchas.rst (FAQ page)

### PR DESCRIPTION
**Pull request by tommyod**
Minor changes to the documentation, specifically `gotchas.rst`, i.e. the FAQ page.

* Function/method references as links.
* Consistent double backticks ` `` ` around Series, DataFrame.
* Minor rephrasing of a sentence or two, added some missing punctuation.
* Replaced `.. ipython::` with `ipython:: python` twice, as the former is not sufficient to make the iPython code render in the docs.

Reviews/comments very welcome.

### CODE REVIEW

**Comment by jreback** [comment]
> Replaced .. ipython:: with ipython:: python twice, as the former is not sufficient to make the iPython code render in the docs.

would welcome this an additional check in the lint.sh

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -22,21 +22,20 @@ Frequently Asked Questions (FAQ)
...
+when calling :meth:`~DataFrame.info`:
 
 .. ipython:: python
 
     dtypes = ['int64', 'float64', 'datetime64[ns]', 'timedelta64[ns]',
               'complex128', 'object', 'bool']
     n = 5000
-    data = dict([ (t, np.random.randint(100, size=n).astype(t))
+    data = dict([(t, np.random.randint(100, size=n).astype(t))
                     for t in dtypes])
```
Comment (jreback) [inline doc/source/gotchas.rst]: can you fix indent here

File: doc/source/gotchas.rst [hunk 2]
```diff
@@ -22,21 +22,20 @@ Frequently Asked Questions (FAQ)
 
 DataFrame memory usage
 ----------------------
-The memory usage of a dataframe (including the index)
-is shown when accessing the ``info`` method of a dataframe. A
-configuration option, ``display.memory_usage`` (see :ref:`options`),
-specifies if the dataframe's memory usage will be displayed when
-invoking the ``df.info()`` method.
+The memory usage of a ``DataFrame`` (including the index) is shown when calling
+the :meth:`~DataFrame.info`. A configuration option, ``display.memory_usage``
```
Comment (jreback) [inline doc/source/gotchas.rst]: can you reference this option directly?

File: doc/source/gotchas.rst [hunk 3]
```diff
@@ -320,7 +323,7 @@ a different byte order than the one on which you are running Python. A common sy
 
 To deal
 with this issue you should convert the underlying NumPy array to the native
-system byte order *before* passing it to Series/DataFrame/Panel constructors
+system byte order *before* passing it to ``Series``/``DataFrame`` constructors
```
Comment (jreback) [inline doc/source/gotchas.rst]: / -> or

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jorisvandenbossche** [comment]
There are certain legitimate use cases of `.. ipython::` blocks with the `python` added, so I am not sure we can add such a lint check in general.

**Comment by jorisvandenbossche** [comment]
For example, the code snippet here is done with a bar `.. ipython::` on purpose: http://pandas.pydata.org/pandas-docs/stable/basics.html#iteritems. I don't recall exactly, but I think it was a problem with the for loop in a ipython::python directive (but it might be that that works correctly by now, but we would need to check).

File: ci/lint.sh [hunk 4]
```diff
@@ -149,6 +149,15 @@ if [ "$LINT" ]; then
         RET=1
     fi
     echo "Check for deprecated messages without sphinx directive DONE"
+
```
Comment (jreback) [inline ci/lint.sh]: does this pass now?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
small question. looks fine. can you rebase and push again just to make sure.

**Comment by jorisvandenbossche** [comment]
@jreback no, this is not ready, see my comment

BTW, can you answer https://github.com/pandas-dev/pandas/issues/19750 before merging this?

**Comment by jorisvandenbossche** [comment]
Ah, I didn't see that the CI was passing. But since we still have occurrences of `.. ipython::` without `python` in our docs, that means the lint is check is actually not working?

**Comment by tommyod** [comment]
I believe this PR is good to go now.

Thanks for double-checking the lint @jorisvandenbossche. It seems like it was not working, glad you caught it. Since there might be cases which warrant `.. ipython::` without `python`, I have removed the lint for now. In the particular case of the code blocks where I added `python` in this PR, the code is [currently not rendered](http://pandas-docs.github.io/pandas-docs-travis/gotchas.html#using-the-in-operator) - so adding `python` seems correct as it now renders on my local doc build.

**Comment by jreback** [comment]
``.. ipython:: without python``

I guess, but why can't we cross this bridge i its actually needed. not-specifying python makes this look not so great and easy enough to avoid with a lint check.

**Comment by jorisvandenbossche** [comment]
The cases of `.. ipython:: without python` that @tommyod fixed, where indeed correct fixes (it were cases where we forgot this, and by forgetting it indeed did not render correctly in the docs). 

But as I mentioned above, there are *other* cases where this was done on purpose (https://github.com/pandas-dev/pandas/pull/19747#issuecomment-366547716). And in those they are currently rendered fine. I am not sure if those cases are actually needed, but if someone wants to remove them, we should first check the docs build correctly without them.

**Comment by jreback** [comment]
ok that’s fine 
better to have automatically checks in things if possible otherwise end up with inconsistencies over time

**Pull request merged**